addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+        apiClient.setTcsMode(tcsId, mode);
+    }
+
+    public void setPermanentSetPoint(String zoneId, double doubleValue) {
+        apiClient.setHeatingZoneOverride(zoneId, doubleValue);
+    }
+
+    public void cancelSetPointOverride(String zoneId) {
+        apiClient.cancelHeatingZoneOverride(zoneId);
+    }
+
+    public void addAccountStatusListener(AccountStatusListener listener) {
+        listeners.add(listener);
+        listener.accountStatusChanged(getThing().getStatus());
+    }
+
+    public void removeAccountStatusListener(AccountStatusListener listener) {
+        listeners.remove(listener);
+    }
+
+    private boolean checkInstallationInfoHasDuplicateIds(Locations locations) {
+        boolean result = true;
+
+        // Make sure that there are no duplicate IDs
+        Set<String> ids = new HashSet<String>();
+
+        for (Location location : locations) {
+            result &= ids.add(location.getLocationInfo().getLocationId());
+            for (Gateway gateway : location.getGateways()) {
+                result &= ids.add(gateway.getGatewayInfo().getGatewayId());
+                for (TemperatureControlSystem tcs : gateway.getTemperatureControlSystems()) {
+                    result &= ids.add(tcs.getSystemId());
+                    for (Zone zone : tcs.getZones()) {
+                        result &= ids.add(zone.getZoneId());
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private void disposeApiClient() {
+        if (apiClient != null) {
+            apiClient.logout();
+        }
+        apiClient = null;
+    }
+
+    private void disposeRefreshTask() {
+        if (refreshTask != null) {
+            refreshTask.cancel(true);
+        }
+    }
+
+    private boolean checkConfig() {
+        String errorMessage = "";
+
+        if (configuration == null) {
+            errorMessage = "Configuration is missing or corrupted";
+        } else if (StringUtils.isEmpty(configuration.username)) {
+            errorMessage = "Username not configured";
+        } else if (StringUtils.isEmpty(configuration.password)) {
+            errorMessage = "Password not configured";
+        } else {
+            return true;
+        }
+
+        updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
+        return false;
+    }
+
+    private void startRefreshTask() {
+        disposeRefreshTask();
+
+        refreshTask = scheduler.scheduleWithFixedDelay(this::update, 0, configuration.refreshInterval,
+                TimeUnit.SECONDS);
+    }
+
+    private void update() {
+        try {
+            apiClient.update();
+            updateAccountStatus(ThingStatus.ONLINE);
+            updateThings();
+        } catch (Exception e) {
+            updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.warn("Failed to update installation status", e);
```

Comment:
reduce to debug, you are already reporting the problem through the thing status.